### ISSUE TITLE
virtual sites mess up numbering, causing issue with bond and order

### ISSUE THREAD

**Issue by brisvag** [mjtadema/garnish#44]
I just discovered that Tryptophan's virtual site (SC3) is causing issues.

Particularly, when trying to guess molecules based on connectivity, the presence of a tryptophan creates new, wrong molecule containing only the virtual site (since this is not bound to anything).

There may be other problems because of this!

**Comment by brisvag** [comment]
gmx dump identifies virtual site like this (first one is normal atom, second one is virtual site):
```
            atom[     5]={type= 24, typeB= 24, ptype=    Atom, m= 7.20000e+01, q= 0.00000e+00, mB= 7.20000e+01, qB= 0.00000e+00, resind=    0, atomnumber= -1}
            atom[  1753]={type=  9, typeB=  9, ptype=   VSite, m= 0.00000e+00, q= 0.00000e+00, mB= 0.00000e+00, qB= 0.00000e+00, resind=  760, atomnumber= -1}
```

**Comment by brisvag** [comment]
As for the solution:
1) add a fake bond to connect the residue to the virtual site. Fixes probably any issues we may have with networkx. However, it may be a bit hard to do this programmatically for any weird topology that may contain virtual sites
2) remove them from the system. Probably bad for many reason, like indexes being messed up
3) mark them as virtual sites in the graph, which probably lets us ignore them when needed.

I would go with 3.

**Comment by brisvag** [comment]
Actually, I think I found the section we needed in the `gmx dump`:
```
            0 type=7991 (VSITEN)  85  87
            1 type=7991 (VSITEN)  85  86
            2 type=7991 (VSITEN)  85  84
```

**Comment by brisvag** [comment]
Finally implemented in 3844a736d97967c1628720b4bb6a3979a00d2634. VSites data can be gathered from `gmx dump` as I said above, but also from topologies from the `virtual_sitesn` section.

Now, the system dictionary also has a `vsiten` section.
